### ISSUE TITLE
Memory leak virtual address space huge

### ISSUE THREAD

**Issue by sumariva** [maharmstone/btrfs#640]
I opening this issue as I suspect that the driver may be leaking memory.
I could only realize for sure if you instruct me, how to compile the code with some file logging support.
If the driver could write down the PID or some other details.
I posting the images about what I think it is.
The system is uptime for 8 days. Just suspended it yesterday, when I flipped the screen.

![btrfs-uptime](https://github.com/maharmstone/btrfs/assets/6371372/a103430b-4363-4324-8849-ccd4ba5f5ca3)
One week of hard work, with crypto, some games, Internet access.

![btrfs-driver-i-guess](https://github.com/maharmstone/btrfs/assets/6371372/82980a16-8498-41c0-8065-21095fd7ea51)
I am not sure if it is the driver. If the driver could write down or have some place to inquire the process id it is running on...

![btrfs-virtual-memory](https://github.com/maharmstone/btrfs/assets/6371372/4d2b51c1-e031-4101-913e-538905010d02)
Memory space using about 20GiB. Noticing system degradation due to large disk swapping on SSD. I will need to reboot shortly.

As I stated, I could experiment with the new driver releases. I have been using this machine all working days, until April 15th.

**Comment by maharmstone** [comment]
If you can attach a kernel debugger and run `!poolused` in windbg, it breaks down the memory usage by driver. The Btrfs tag is MHBt.

>Paged pool: 5.6 GB
>Non-paged pool: 5.4 GB

These are both higher than I'd expect. If there was a memory leak in the driver, I'd expect one or other of them to be high.

Also, are you using compression? I've not fully audited the zlib, lzo, and zstd compression libraries we're using.

**Comment by sumariva** [comment]
Yes, I am using a lot of zstd compression on directories that store the databases for the crypto currencies.
Severe random file access for read and writes at tails of database.
I also sent a copy of monero database, 65GiB in compact mode, so all these was read from btrfs with zstd and written compressed to a NTFS with LZMA. The compressed package was sent through network and reached friend's computer. Looks like in perfect state.

Just to mention, the LZMA compression, which was made by 7-zip tool, was very superior to zstd. If those compressions are things available to add on on btrfs, this one is a very good one!
